tests/test_profiler.py

```diff
@@ -14,34 +14,18 @@
 class TestProfiler(unittest.TestCase):
 
     def test_profiler(self):
-        """Make sure that `starfish --profile noop works."""
+        """Make sure that `starfish --profile --noop works."""
         cmdline = [
             "starfish",
             "--profile",
-            "noop",
+            "--noop",
         ]
-        if cmdline[0] == 'starfish':
-            coverage_cmdline = [
-                "coverage", "run",
-                "-p",
-                "--source", "starfish",
-                "-m", "starfish",
-            ]
-            coverage_cmdline.extend(cmdline[1:])
-            cmdline = coverage_cmdline
         env = os.environ.copy()
-        env[PROFILER_NOOP_ENVVAR] = ""
-        subprocess.check_call(cmdline, env=env)
-
-    def test_noop_hiding(self):
-        """
-        Ensure we do not expose the noop command unless the magic environment flag is present.  This is to avoid
-        confusion from the end users.
-        """
-        cmdline = [
-            "starfish",
-            "noop",
-        ]
+        if sys.version_info < (3, 0):
+            # this is a hideous hack because argparse is not consistent between 2.7 and 3.x about whether the subcommand
+            # is required.  this entire hack can be dropped if we deprecate python2.7.
```

Comment:
I'm OK deprecating python 2.7 -- especially if we keep having to put weird hacks in like this. What do you think?

Reply:
I think dropping python2.7 is fine for the CLI.  It's much less clear whether we should drop python2.7 for API because people might have legacy 2.7 codebases that they want to integrate with starfish.  However, you know far more whether this is likely, and if you're okay, then YAY!!!